ui/src/timeMachine/utils/insertFunction.ts

```diff
@@ -14,13 +14,18 @@ const functionRequiresNewLine = (funcName: string): boolean => {
 
 export const formatFunctionForInsert = (
   funcName: string,
-  fluxFunction: string
-) => {
+  fluxFunction: string,
+  newLine: boolean
```

Comment:
Ya, it's only used in this one place, but the reason for that added boolean is to handle specific edge cases in cases where the cursor position is at the default position (1:1), or on the last line. I hear what you're saying, but besides creating a specific function that would be called conditionally based on a boolean, I'm not sure what other solution there would be. And since conditionally calling a function seems just as fishy to me, I'd love to hear your thoughts

Reply:
i'd like to pair up to talk about this, but my calendar just filled up today so i have to type this out.

>I'm not sure what other solution there would be.

the solution is to look at the design of the software as a whole and change the pieces that don't make sense.

right now, there is logic split between the component and this function in utils. `formatFunctionForInsert` can't operate without external logic to tell it whether it should have newlines or not - that logic, in the form of an argument, is handled elsewhere. it makes the function tightly coupled to the code that calls it. what's the point of having a function, which is meant to promote encapsulation and reuse, if you're not encapsulating or enabling reuse? it just becomes a misdirection that you have to look up.

the solution is to consolidate the logic. either keep the logic in the component the way it is now and remove these methods, or make helper methods (which imply reuse) which contain the logic of figuring out where to put the flux function. if it was me, i'd write some helper methods in the component and leave them there until something else needs to use them.